frontend/src/static/js/v_summary.js

```diff
@@ -305,38 +324,83 @@ window.vSummary = {
       };
       this.filtered = this.sortFiltered(this.filtered, filterControl);
 
-      if (this.isMergeGroup) {
-        this.mergeGroup(this.filtered);
+      const mergeGroupStatus = [];
+      if (this.mergedGroups === 'all' || this.mergedGroups === 'none') {
+        for (let i = 0; i < this.filtered.length; i += 1) {
+          mergeGroupStatus.push(this.mergedGroups === 'all');
+        }
+      } else {
+        const mergedGroups = this.mergedGroups.split(',');
+        for (let i = 0; i < this.filtered.length; i += 1) {
+          mergeGroupStatus.push(mergedGroups.includes(this.getGroupName(this.filtered[i])));
+        }
       }
+
+      mergeGroupStatus.forEach((isMergeGroup, groupIndex) => {
+        if (isMergeGroup) {
+          this.mergeGroupByIndex(this.filtered, groupIndex);
+        }
+      });
     },
 
     mergeGroup(filtered) {
       filtered.forEach((group, groupIndex) => {
-        const dateToIndexMap = {};
```

Comment:
following the suggestion previously, you just need to add a check here if the current group name within `group name -> isMerge` leads to false, if false then you skip the merging by `continue;`. You don't need to additionally create the `mergeGroupByIndex` method.